Examples/Algorithms/Digitization/src/DigitizationOptions.cpp

```diff
@@ -0,0 +1,234 @@
+    auto types = variables["digi-smear-types"].as<read_series>();
+    auto parameters = variables["digi-smear-parameters"].as<read_range>();
+
+    // Count the number of needed parameters
+    // - type 0 needs 1 parameters
+    // - type 1-4 needs 4 parameters:
+    //      1 principle parameter, 2 range parameters, 1 range indicator
+    size_t sumpars = 0;
+    std::for_each(types.begin(), types.end(),
+                  [&](int n) { sumpars += (n == 0) ? 1 : 4; });
+
+    if (volumes.size() == vdims.size() and bvalues.size() == types.size() and
+        parameters.size() == sumpars) {
+      size_t vpos = 0;
+      size_t ppos = 0;
+
+      ACTS_DEBUG("Volume parameters properly read from command line.")
+
+      /// Extract the parameters from the global vector
+      ///
+      /// @param ftype the function type
+      ///
+      /// @return an extracted vector
+      auto extract = [&](int ftype) -> std::vector<double> {
+        size_t padd = (ftype == 0) ? 1 : 3;
+        std::vector<double> fpars = {parameters.begin() + ppos,
+                                     parameters.begin() + ppos + padd};
+        // unfortunately boost::program_options can not deal with negative
+        // values within a sequence, we thus have to flag the range at defined
+        // with command line options:
+        //    < 0 (both negative), 0 (symmetric), > 0 both positive
+        if (padd == 3) {
+          double rangeIndic = parameters[ppos + padd];
+          fpars[1] *= (rangeIndic <= 0) ? -1. : 1.;
+          fpars[2] *= (rangeIndic < 0) ? -1. : 1.;
+          ++padd;
+        }
+        ppos += padd;
+        return fpars;
+      };
+
+      for (unsigned int iv = 0; iv < volumes.size(); ++iv) {
+        SmearingAlgorithm::SupportedSmearer smearer;
+
+        int volID = volumes[iv];
+        Acts::GeometryIdentifier volumeGeometryId =
+            Acts::GeometryIdentifier(0).setVolume(volID);
+
+        int volDim = vdims[iv];
+        int ftype = 0;
+        // 1 - dimensional, either loc 0 or loc 1
+        if (volDim == 1) {
+          using OneFnc = std::array<ActsFatras::SmearFunction<RandomEngine>, 1>;
+          std::array<int, 1> ftypes;
+          std::array<std::vector<double>, 1> fparameters;
+          ftype = types[vpos++];
+          ftypes[0] = ftype;
+          fparameters[0] = extract(ftype);
+          OneFnc smearFunctions = sFnc.generate<1>(ftypes, fparameters);
+          // Create and fill
+          if ((Acts::BoundIndices)bvalues[vpos - 1] == Acts::eBoundLoc0) {
+            smearCfg.configured = true;
+            smearer =
+                std::pair<ActsFatras::BoundParametersSmearer<Acts::eBoundLoc0>,
+                          OneFnc>(
+                    ActsFatras::BoundParametersSmearer<Acts::eBoundLoc0>(),
+                    std::move(smearFunctions));
+          } else if ((Acts::BoundIndices)bvalues[vpos - 1] ==
+                     Acts::eBoundLoc0) {
```

Comment:
I guess the `Acts::eBoundLoc0` should be `Acts::eBoundLoc1` here?

Reply:
Of course you are right.